qdax/baselines/td3.py

```diff
@@ -234,35 +275,83 @@ def eval_policy_fn(
         environments.
 
         Args:
-            policy_params: Params of the NN defining the policy.
-            random_key: a random key
+            training_state: TD3 training state.
             eval_env_first_state: the first state of the environment.
             play_step_fn: function defining how to play a step in the env.
 
         Returns:
             true return averaged over batch dimension, shape: (1,)
             true return per env, shape: (env_batch_size,)
         """
-
-        state, transitions = generate_unroll(
+        # TODO: this generate unroll shouldn't take a random key
+        state, training_state, transitions = generate_unroll(
             init_state=eval_env_first_state,
-            policy_params=policy_params,
-            random_key=random_key,
+            training_state=training_state,
             episode_length=self._config.episode_length,
             play_step_fn=play_step_fn,
         )
 
-        eval_metrics_key = CompletedEvalWrapper.STATE_INFO_KEY
```

Comment:
Same question @Egiob (completely related to the question for sac, just pinging here as well to be sure we do not forget)